### ISSUE TITLE
Android: Error deleting cookie whose name starts with "session"

### ISSUE THREAD

**Issue by MicahLC** [appium/appium#11052]
## The problem

When I try to delete a cookie named `session-id` on Android, I get this exception:
```
An unknown server-side error occurred while processing the command. Original error: Could not proxy. Proxy error: Could not find :session section for url: /session2-id-longer-really-long-name (WARNING: The server did not provide any stacktrace information)
```
This happens for any cookie that starts with `session`, like `session-id2` or `session2-id`. Cookies named `session` give a slightly different but similar error.

However, if I use `driver.manage().deleteAllCookies()`, the cookie is deleted without error.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.8.1
* Desktop OS/version used to run Appium: OS X Sierra
* Mobile platform/version under test: Android 8.0
* Real device or emulator/simulator: Emulator (although it happens on Sauce Labs and Browser Stack with real devices, too)
* Appium CLI or Appium.app|exe: Appium.app

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

This log is from when I ran it locally:
https://gist.github.com/MicahLC/a058b3d07ba494c2912506b33e22f0eb

## Code To Reproduce Issue [ Good To Have ]

https://github.com/MicahLC/android_session_cookie_bug

I had already created this repro case by the time I saw the instructions about the `sample-code` directory. Please let me know if this repro case isn't usable and I'll update it to use the sample-code.

**Comment by mykola-mokhnach** [comment]
@CalebRouleau FYI

**Comment by mykola-mokhnach** [comment]
This is just my assumption. I didn't dig into their code too deep. Although, it might be that if urls matching logic is buggy and delete by name endpoint includes the whole delete all cookies endpoint (`session/:sessionId/cookie` -> `session/:sessionId/cookie/:name`) then chromedriver matches incorrect handler and throws the error above.

**Comment by MicahLC** [comment]
OK. I was asking because the line about proxying at https://gist.github.com/MicahLC/a058b3d07ba494c2912506b33e22f0eb#file-appium-log-from-ui-L219 looks awfully suspicious. It looks like it's cutting off everything before the second instance of `session` in the string. However, I don't know enough about the code involved to know whether that's an appium issue or a chromium issue. Where is the proxying coming from? Are there rules for that defined somewhere?

**Comment by imurchie** [comment]
Yes, that looks like a problem on Appium's side. Give me a moment to look into it.

**Comment by imurchie** [comment]
The problem was in our parsing of the url when proxying. See https://github.com/appium/appium-base-driver/pull/236
